src/ape/managers/query.py

```diff
@@ -1,13 +1,42 @@
-from typing import Dict, Optional
+from eth_utils import keccak, to_hex, to_bytes
+from functools import partial
+from typing import Any, Dict, Optional
 
 import pandas as pd
+from pydantic import BaseModel
 
 from ape.api import QueryAPI, QueryType
+from ape.api.query import ContractEventQuery, BlockQuery, _BaseQuery
 from ape.exceptions import QueryEngineError
 from ape.plugins import clean_plugin_name
 from ape.utils import ManagerAccessMixin, cached_property
 
 
+def get_columns_from_item(query: _BaseQuery, item: BaseModel) -> Dict[str, Any]:
+    return {k: v for k, v in item.dict().items() if k in query.columns}
+
+
+class DefaultQueryProvider(QueryAPI):
+    def estimate_query(self, query: QueryType) -> Optional[int]:
+        if isinstance(query, BlockQuery):
+            # NOTE: Very loose estimate of 100ms per block
+            return (query.stop_block - query.start_block) * 100
```

Comment:
Are events going to have different multiplier values than blocks etc?
How are you choosing values?

Reply:
Just a loosely-chosen heuristic that is more-or-less accurate, so that the selection algorithm works correctly when other query plugins are added that can fulfill the query faster